### PR TITLE
SF-2362 Upgrade legacy mat-dialog to MDC version

### PR DESCRIPTION
**Pull request by RaymondLuong3**
This PR updates the legacy mat-dialog to the MDC based version. This used `ng generate @angular/material:mdc-migration` as a starting point.
Reviewers should test the components thoroughly. This will require tester to test this as it affects a large number of components.

- - -
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2489)

### CODE REVIEW

**Review by nigel-wells — CHANGES_REQUESTED** [review]
This is a massive undertaking - thanks for working through it all.

I'm  just going to mention a few instances of things that look odd as we'll probably need to address similar things across all dialogs.

Sometimes the action buttons don't have the same bottom/right padding as we currently do. Is this a case of the new material way which we should support and ensure all our margins are the same?


Some dialogs have quite different line height. Is that the new material way or something else? I wouldn't expect our notice component to change because its parent component changed either.


> Reviewed 90 of 90 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2489#-:-NyxPVBJ5dBj52nA4q9X:bxugo4a)* status: all files reviewed, 6 unresolved discussions (waiting on @RaymondLuong3)

___
*[`src/SIL.XForge.Scripture/ClientApp/src/material-styles.scss` line 13 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2489#-NyxHAwzBT7vxAZGgbnM:-NyxHAwzBT7vxAZGgbnN:b-jenmma) ([raw file](https://github.com/sillsdev/web-xforge/blob/a87ca205df341986a7898e903b4f39c14e3f2d5d/src/SIL.XForge.Scripture/ClientApp/src/material-styles.scss#L13)):*
> ```SCSS
> //  If you don't need the default component typographies but still want the hierarchy styles,
> //  you can delete this line and instead use:
> //    `@include mat.legacy-typography-hierarchy(mat.define-legacy-typography-config());`
> ```

Was this comment supposed to change?
___
*[`src/SIL.XForge.Scripture/ClientApp/src/material-styles.scss` line 17 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2489#-NyxHIiO-02vK7jv2QxE:-NyxHIiP7LUTN4FZ6MJl:b-zdx7xq) ([raw file](https://github.com/sillsdev/web-xforge/blob/46fbe32c09dd796aaaf85097540743e4cff86180/src/SIL.XForge.Scripture/ClientApp/src/material-styles.scss#L17)):*
> ```SCSS
> @include mat.all-legacy-component-typographies();
> @include mat.all-component-typographies();
> @include mat.all-component-typographies();
> ```

Double up include?
___
*[`src/SIL.XForge.Scripture/ClientApp/src/app/checking/question-dialog/question-dialog.component.html` line 3 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2489#-NyxLOey84oTWKaG-6Ql:-NyxLOey84oTWKaG-6Qm:bu6t19b) ([raw file](https://github.com/sillsdev/web-xforge/blob/46fbe32c09dd796aaaf85097540743e4cff86180/src/SIL.XForge.Scripture/ClientApp/src/app/checking/question-dialog/question-dialog.component.html#L3)):*
> ```HTML
> <ng-container *transloco="let t; read: 'question_dialog'">
>   <div class="dialog-container">
>     <h1 mat-dialog-title class="dialog-title dialog-icon-title">
> ```

Just noting this as an observation. The new `dialog-icon-title` was to fix the height above the text and icon. I can see the solution being useful for headings with a button specific icon but not an icon on its own like this dialog. In this case the issue is only caused by the title now being `display: flex` which then disrupts the `:before` style which was what was providing the height above the text and icon.
___
*[`src/SIL.XForge.Scripture/ClientApp/src/app/shared/share/share-dialog.component.scss` line 43 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2489#-NyxNXED817GfW1bcBZP:-NyxNXED817GfW1bcBZQ:b-xhq7ix) ([raw file](https://github.com/sillsdev/web-xforge/blob/a87ca205df341986a7898e903b4f39c14e3f2d5d/src/SIL.XForge.Scripture/ClientApp/src/app/shared/share/share-dialog.component.scss#L43)):*
> ```SCSS
> div[mat-dialog-actions] {
>   display: flex;
>   flex-direction: row-reverse; /* This was done so the share button would receive the focus */
> ```

Initially I was wondering why this was removed (along with the buttons changing in the HTML) but I see it wasn't even doing what I thought it was supposed to be doing on live which was focusing on the share button. It must have broke at some point. I can't actually recall the "why" behind doing it - I think it was to allow a user to quickly press enter on their keyboard rather than click the button. It was probably adding code that didn't need to be there so I'm happy for it to be reverted and made simplier.
___
*[`src/SIL.XForge.Scripture/ClientApp/src/app/translate/editor/note-dialog/note-dialog.component.html` line 2 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2489#-NyxOfIa2LbF6KvyLr8I:-NyxOfIa2LbF6KvyLr8J:b-1gb32q) ([raw file](https://github.com/sillsdev/web-xforge/blob/46fbe32c09dd796aaaf85097540743e4cff86180/src/SIL.XForge.Scripture/ClientApp/src/app/translate/editor/note-dialog/note-dialog.component.html#L2)):*
> ```HTML
> <ng-container *transloco="let t; read: 'note_dialog'">
>   <div class="dialog-container">
> ```

Why does this dialog need another container and the others don't?
___
*[`src/SIL.XForge.Scripture/ClientApp/src/xforge-common/feature-flags/feature-flags-dialog.component.scss` line 16 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2489#-NyxP27CCRETQ4F2nm37:-NyxP27CCRETQ4F2nm38:bdnk7ig) ([raw file](https://github.com/sillsdev/web-xforge/blob/a87ca205df341986a7898e903b4f39c14e3f2d5d/src/SIL.XForge.Scripture/ClientApp/src/xforge-common/feature-flags/feature-flags-dialog.component.scss#L16)):*
> ```SCSS
> app-notice {
>   max-width: 25em;
>   margin: 1em 0;
> ```

Would be nice to still have `margin-bottom: 1em`

**Review by RaymondLuong3 — COMMENTED** [review]
The action buttons do look rather tight to the corner. I've added padding to make it look more as it did before. Looking at the dialogs on the material.angular.io site, there appears to be plenty of space around the action buttons.

The share dialog was a little interesting. It was not using the `mat-dialog-content` directive at all, so the padding gets lost completely with the update. The documentation for the v16 dialog does explain the changes about the extra roomy text, and I keep flip-flopping on whether I like it, or if I like the original more. [Migrating to MDC-based Components | Angular Material](https://material.angular.io/guide/mdc-migration#dialog)

Here it is with the custom padding and without the `mat-dialog-content` directive.


> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2489#-:-Nz-8rIH5aa9fmK84iP0:bat8d42)* status: all files reviewed, 4 unresolved discussions (waiting on @nigel-wells)

___
*[`src/SIL.XForge.Scripture/ClientApp/src/material-styles.scss` line 13 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2489#-NyxHAwzBT7vxAZGgbnM:-Nz-2B1S4Xg513jfokbn:bvup1mw) ([raw file](https://github.com/sillsdev/web-xforge/blob/a87ca205df341986a7898e903b4f39c14e3f2d5d/src/SIL.XForge.Scripture/ClientApp/src/material-styles.scss#L13)):*
<details><summary><i>Previously, nigel-wells (Nigel Wells) wrote…</i></summary><blockquote>

Was this comment supposed to change?
</blockquote></details>

Yes, I think it was supposed to change. This was from the `ng generate` command.
___
*[`src/SIL.XForge.Scripture/ClientApp/src/material-styles.scss` line 17 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2489#-NyxHIiO-02vK7jv2QxE:-Nz-2Ksm84Bo5hT_7sDs:b-73m6vw) ([raw file](https://github.com/sillsdev/web-xforge/blob/46fbe32c09dd796aaaf85097540743e4cff86180/src/SIL.XForge.Scripture/ClientApp/src/material-styles.scss#L17)):*
<details><summary><i>Previously, nigel-wells (Nigel Wells) wrote…</i></summary><blockquote>

Double up include?
</blockquote></details>

Nice catch!
___
*[`src/SIL.XForge.Scripture/ClientApp/src/app/checking/question-dialog/question-dialog.component.html` line 3 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2489#-NyxLOey84oTWKaG-6Ql:-Nz-2k6YAww-Yik6AO91:b-xsjqng) ([raw file](https://github.com/sillsdev/web-xforge/blob/46fbe32c09dd796aaaf85097540743e4cff86180/src/SIL.XForge.Scripture/ClientApp/src/app/checking/question-dialog/question-dialog.component.html#L3)):*
<details><summary><i>Previously, nigel-wells (Nigel Wells) wrote…</i></summary><blockquote>

Just noting this as an observation. The new `dialog-icon-title` was to fix the height above the text and icon. I can see the solution being useful for headings with a button specific icon but not an icon on its own like this dialog. In this case the issue is only caused by the title now being `display: flex` which then disrupts the `:before` style which was what was providing the height above the text and icon.
</blockquote></details>

Good point. It is too bad that `mat-dialog-title` does not natively support icons and buttons as part of the title so we would not need `dialog-icon-title` at all.
___
*[`src/SIL.XForge.Scripture/ClientApp/src/app/shared/share/share-dialog.component.scss` line 43 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2489#-NyxNXED817GfW1bcBZP:-Nz-3E6F0IgJWUDP3VFF:b2ixlji) ([raw file](https://github.com/sillsdev/web-xforge/blob/a87ca205df341986a7898e903b4f39c14e3f2d5d/src/SIL.XForge.Scripture/ClientApp/src/app/shared/share/share-dialog.component.scss#L43)):*
<details><summary><i>Previously, nigel-wells (Nigel Wells) wrote…</i></summary><blockquote>

Initially I was wondering why this was removed (along with the buttons changing in the HTML) but I see it wasn't even doing what I thought it was supposed to be doing on live which was focusing on the share button. It must have broke at some point. I can't actually recall the "why" behind doing it - I think it was to allow a user to quickly press enter on their keyboard rather than click the button. It was probably adding code that didn't need to be there so I'm happy for it to be reverted and made simplier.
</blockquote></details>

I agree, the behaviour didn't seem to be necessary, so it is nice to keep it simple again.
___
*[`src/SIL.XForge.Scripture/ClientApp/src/app/translate/editor/note-dialog/note-dialog.component.html` line 2 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2489#-NyxOfIa2LbF6KvyLr8I:-Nz-3PvvE6dxtQyLb4PH:bxftbqe) ([raw file](https://github.com/sillsdev/web-xforge/blob/46fbe32c09dd796aaaf85097540743e4cff86180/src/SIL.XForge.Scripture/ClientApp/src/app/translate/editor/note-dialog/note-dialog.component.html#L2)):*
<details><summary><i>Previously, nigel-wells (Nigel Wells) wrote…</i></summary><blockquote>

Why does this dialog need another container and the others don't?
</blockquote></details>

The textbox inside the note dialog was causing overflow and an ugly scrollbar to appear. So I made a new div and made its display to be flex so that the unnecessary scroll bar would go away. Here is a picture of what it looked like before adding the extra div.

___
*[`src/SIL.XForge.Scripture/ClientApp/src/xforge-common/feature-flags/feature-flags-dialog.component.scss` line 16 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2489#-NyxP27CCRETQ4F2nm37:-Nz-52ge9-Hjjfugy8b_:b-896fix) ([raw file](https://github.com/sillsdev/web-xforge/blob/a87ca205df341986a7898e903b4f39c14e3f2d5d/src/SIL.XForge.Scripture/ClientApp/src/xforge-common/feature-flags/feature-flags-dialog.component.scss#L16)):*
<details><summary><i>Previously, nigel-wells (Nigel Wells) wrote…</i></summary><blockquote>

Would be nice to still have `margin-bottom: 1em`
</blockquote></details>

Done.

File: src/SIL.XForge.Scripture/ClientApp/src/material-styles.scss [hunk 1]
```diff
@@ -121,7 +123,7 @@ $material-theme-accent-error: mat.define-light-theme(
 @include mat.card-theme($material-app-theme);
 @include mat.checkbox-theme($material-app-theme);
 @include mat.chips-theme($material-app-theme);
-@include mat.legacy-dialog-theme($material-app-theme);
+@include mat.dialog-theme($material-app-theme);
```
Comment (Nateowami) [inline src/SIL.XForge.Scripture/ClientApp/src/material-styles.scss:126]: The text on the dialog ends up looking very faded. I think this is controlled by the theme we pass here. We may need to adjust the theme we pass here so the text color isn't so light. It's much darker in the examples at https://material.angular.io/components/dialog/examples

Also, please make sure you're reading through the upgrade information at https://material.angular.io/guide/mdc-migration#dialog

**Review by nigel-wells — CHANGES_REQUESTED** [review]
Thanks for the link. I find it interesting that they say you don't have to use the `mat-dialog-content` if you don't want to. I'm sure there are use cases but we'd get better consistency, and less effort, if all our dialogs used `mat-dialog-content` and `mat-dialog-actions`. It would also give us less to maintain as Material is updated in the future which is exactly the issue we're finding now.

We do need to implement our material theme better - it currently only has a couple of color palettes. You could add in some basic typography which resolves the line out issue.

**material-styles.scss**

```
$app-typography: mat.define-typography-config(
  $body-1: mat.define-typography-level(16px, 1.2em)
);

$material-app-theme: mat.define-light-theme(
  (
    color: (
      primary: $matGreyPalette,
      accent: $matGreenLightPalette,
      warn: $matErrorPalette
    ),
    typography: $app-typography
  )
);
```

The lighter color is the dialog adding a `0.6` transparency on the color. I'm sure this can be overwritten in the theme setup as well.

> Reviewed 1 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2489#-:-Nz1JJ298OeguRXwI5re:b-891j2r)* status: 87 of 90 files reviewed, 2 unresolved discussions (waiting on @RaymondLuong3)

___
*[`src/SIL.XForge.Scripture/ClientApp/src/app/translate/editor/note-dialog/note-dialog.component.html` line 2 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2489#-NyxOfIa2LbF6KvyLr8I:-Nz1Qfy74QqoR9YLuTMS:b-ubtetl) ([raw file](https://github.com/sillsdev/web-xforge/blob/46fbe32c09dd796aaaf85097540743e4cff86180/src/SIL.XForge.Scripture/ClientApp/src/app/translate/editor/note-dialog/note-dialog.component.html#L2)):*
<details><summary><i>Previously, RaymondLuong3 (Raymond Luong) wrote…</i></summary><blockquote>

The textbox inside the note dialog was causing overflow and an ugly scrollbar to appear. So I made a new div and made its display to be flex so that the unnecessary scroll bar would go away. Here is a picture of what it looked like before adding the extra div.

</blockquote></details>

Can this be looked at again with the other comments taken into account around having consistent dialog elements for container and buttons

**Review by RaymondLuong3 — COMMENTED** [review]
Thanks for the suggestion. It worked like a charm. I did find that for the share dialog and bulk import dialog, the 16px text appeared larger than before. It was 14px on those dialogs.
I added an extra topography configuration for dialogs. This made all other dialogs with the 16px text smaller, but to me it looks OK. Let me know if you would rather if I applied a custom font-size to the dialogs on a case by case basis.

> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2489#-:-Nz4vTxm24VfuAoUkM6Q:br60oec)* status: 87 of 90 files reviewed, 2 unresolved discussions (waiting on @Nateowami and @nigel-wells)

___
*[`src/SIL.XForge.Scripture/ClientApp/src/material-styles.scss` line 126 at r2](https://reviewable.io/reviews/sillsdev/web-xforge/2489#gh-1617919133:-Nz4wrreExXpAhzbzaPa:b-mchu5g) ([raw file](https://github.com/sillsdev/web-xforge/blob/77ff65c21c514ab59810b4c27e0df5eed1ae61e9/src/SIL.XForge.Scripture/ClientApp/src/material-styles.scss#L126)):*
<details><summary><i>Previously, Nateowami (Nathaniel Paulus) wrote…</i></summary><blockquote>

The text on the dialog ends up looking very faded. I think this is controlled by the theme we pass here. We may need to adjust the theme we pass here so the text color isn't so light. It's much darker in the examples at https://material.angular.io/components/dialog/examples

Also, please make sure you're reading through the upgrade information at https://material.angular.io/guide/mdc-migration#dialog



</blockquote></details>

Yes, the faded text can be adjusted. I updated the css var that controls dialog text colour. It was not clear how to do it using the angular theming.
___
*[`src/SIL.XForge.Scripture/ClientApp/src/app/translate/editor/note-dialog/note-dialog.component.html` line 2 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2489#-NyxOfIa2LbF6KvyLr8I:-Nz4xIIlAYthb7w5GSnZ:b-quuf9w) ([raw file](https://github.com/sillsdev/web-xforge/blob/46fbe32c09dd796aaaf85097540743e4cff86180/src/SIL.XForge.Scripture/ClientApp/src/app/translate/editor/note-dialog/note-dialog.component.html#L2)):*
<details><summary><i>Previously, nigel-wells (Nigel Wells) wrote…</i></summary><blockquote>

Can this be looked at again with the other comments taken into account around having consistent dialog elements for container and buttons
</blockquote></details>

The dialog looks much the same as it did before. I don't think it has to do with `mat-dialog-content` which was already present in the html. The text box was causing the overflow. The simplest way I found to improve this is by updating the html and css rules.

**Review by nigel-wells — CHANGES_REQUESTED** [review]
Glad it helped. I think if we can get as close to "normal" as we can with this one that would be great. After all he legacy stuff is removed we need a task to improve our angular theme support which will make our lives much easier going forward I think.

> Reviewed 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2489#-:-Nz6p3oFCA24sFYg0szv:bemkjo2)* status: all files reviewed, 2 unresolved discussions (waiting on @Nateowami and @RaymondLuong3)

___
*[`src/SIL.XForge.Scripture/ClientApp/src/app/translate/editor/note-dialog/note-dialog.component.html` line 2 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2489#-NyxOfIa2LbF6KvyLr8I:-Nz6qyeB19gM_9YBUuUZ:b1xet1j) ([raw file](https://github.com/sillsdev/web-xforge/blob/46fbe32c09dd796aaaf85097540743e4cff86180/src/SIL.XForge.Scripture/ClientApp/src/app/translate/editor/note-dialog/note-dialog.component.html#L2)):*
<details><summary><i>Previously, RaymondLuong3 (Raymond Luong) wrote…</i></summary><blockquote>

The dialog looks much the same as it did before. I don't think it has to do with `mat-dialog-content` which was already present in the html. The text box was causing the overflow. The simplest way I found to improve this is by updating the html and css rules.
</blockquote></details>

Ok. Are you able to add the bottom padding in to be consistent with other dialogs - I realise this is already an issue on live but hopefully easy to address as part of this dialog PR.

**Review by RaymondLuong3 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2489)* status: all files reviewed, 2 unresolved discussions (waiting on @Nateowami and @nigel-wells)

___
*[`src/SIL.XForge.Scripture/ClientApp/src/app/translate/editor/note-dialog/note-dialog.component.html` line 2 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2489#-NyxOfIa2LbF6KvyLr8I:-Nz9Ki3045d7dw1wVT4F:bk80gee) ([raw file](https://github.com/sillsdev/web-xforge/blob/46fbe32c09dd796aaaf85097540743e4cff86180/src/SIL.XForge.Scripture/ClientApp/src/app/translate/editor/note-dialog/note-dialog.component.html#L2)):*
<details><summary><i>Previously, nigel-wells (Nigel Wells) wrote…</i></summary><blockquote>

Ok. Are you able to add the bottom padding in to be consistent with other dialogs - I realise this is already an issue on live but hopefully easy to address as part of this dialog PR.
</blockquote></details>

Yes, there is padding that is defined across all dialogs with action buttons. Here is the current view of the note dialog.

**Review by nigel-wells — APPROVED** [review]
:lgtm:

> Reviewed 5 of 5 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2489#-:-NzBFUNa5VafDj9BGYlu:bnfp4nl)* status: all files reviewed, 1 unresolved discussion (waiting on @Nateowami)

___
*[`src/SIL.XForge.Scripture/ClientApp/src/app/translate/editor/note-dialog/note-dialog.component.html` line 2 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2489#-NyxOfIa2LbF6KvyLr8I:-NzBFGU9AprZE62VsBHF:b-bdgia3) ([raw file](https://github.com/sillsdev/web-xforge/blob/46fbe32c09dd796aaaf85097540743e4cff86180/src/SIL.XForge.Scripture/ClientApp/src/app/translate/editor/note-dialog/note-dialog.component.html#L2)):*
<details><summary><i>Previously, RaymondLuong3 (Raymond Luong) wrote…</i></summary><blockquote>

Yes, there is padding that is defined across all dialogs with action buttons. Here is the current view of the note dialog.

</blockquote></details>

You're right - I was on an older commit :)

**Comment by Nateowami** [comment]
@RaymondLuong3 I've left some comments in Chromatic, which unlike Reviewable don't get posted to the PR itself.

**Comment by RaymondLuong3** [comment]
Thanks, I've made the adjustments to the dialogs that you pointed out in chromatic.

**Review by Nateowami — DISMISSED** [review]
> Reviewed 4 of 5 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2489)* status: 90 of 91 files reviewed, 1 unresolved discussion (waiting on @nigel-wells and @RaymondLuong3)

___
*[`src/SIL.XForge.Scripture/ClientApp/src/app/checking/import-questions-dialog/import-questions-confirmation-dialog/import-questions-confirmation-dialog.component.scss` line 3 at r5](https://reviewable.io/reviews/sillsdev/web-xforge/2489#-NzZT0qt4xDacZJe073j:-NzZT0qt4xDacZJe073k:btxvdkc) ([raw file](https://github.com/sillsdev/web-xforge/blob/65d8ee8f3006e25b78d1c4c0d1e201efe7f80580/src/SIL.XForge.Scripture/ClientApp/src/app/checking/import-questions-dialog/import-questions-confirmation-dialog/import-questions-confirmation-dialog.component.scss#L3)):*
> ```SCSS
> .mat-mdc-dialog-title {
>   &:before {
>     display: none;
> ```

I don't think this is the right solution. As I mentioned in the comment in Chromatic (perhaps I wasn't clear enough), this spacing is caused by white space that is affecting the rendering. You just need to make this change to the template:
``` diff
-    <h2 mat-dialog-title>
-      {{ t("some_questions_edited") }}
-    </h2>
+    <h2 mat-dialog-title>{{ t("some_questions_edited") }}</h2>
```

Then there's no need for any CSS change.
___
*[`src/SIL.XForge.Scripture/ClientApp/src/xforge-common/generic-dialog/generic-dialog.component.ts` line 34 at r5](https://reviewable.io/reviews/sillsdev/web-xforge/2489#-NzZWRaN2-Xpcmv2gPQ_:-NzZWRaN2-Xpcmv2gPQa:b-d07yy3) ([raw file](https://github.com/sillsdev/web-xforge/blob/65d8ee8f3006e25b78d1c4c0d1e201efe7f80580/src/SIL.XForge.Scripture/ClientApp/src/xforge-common/generic-dialog/generic-dialog.component.ts#L34)):*
> ```XML
>   }
> 
>   get message$(): Observable<string> | undefined {
> ```

Thanks for fixing these.

**Review by RaymondLuong3 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2489)* status: 90 of 92 files reviewed, 1 unresolved discussion (waiting on @Nateowami and @nigel-wells)

___
*[`src/SIL.XForge.Scripture/ClientApp/src/app/checking/import-questions-dialog/import-questions-confirmation-dialog/import-questions-confirmation-dialog.component.scss` line 3 at r5](https://reviewable.io/reviews/sillsdev/web-xforge/2489#-NzZT0qt4xDacZJe073j:-NzZxfD4BidfLljs7IYS:b-896fix) ([raw file](https://github.com/sillsdev/web-xforge/blob/65d8ee8f3006e25b78d1c4c0d1e201efe7f80580/src/SIL.XForge.Scripture/ClientApp/src/app/checking/import-questions-dialog/import-questions-confirmation-dialog/import-questions-confirmation-dialog.component.scss#L3)):*
<details><summary><i>Previously, Nateowami (Nathaniel Paulus) wrote…</i></summary><blockquote>

I don't think this is the right solution. As I mentioned in the comment in Chromatic (perhaps I wasn't clear enough), this spacing is caused by white space that is affecting the rendering. You just need to make this change to the template:
``` diff
-    <h2 mat-dialog-title>
-      {{ t("some_questions_edited") }}
-    </h2>
+    <h2 mat-dialog-title>{{ t("some_questions_edited") }}</h2>
```

Then there's no need for any CSS change.
</blockquote></details>

Done.

**Review by Nateowami — APPROVED** [review]
> Reviewed 5 of 5 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2489)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @nigel-wells)

**Pull request merged**